### PR TITLE
Delete items both the client-side and Firestore Database

### LINKED ISSUE #11: 11. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future.
## Summary
Users might make a mistake when entering an item, or may decide they won’t be buying a certain item again in the future. Allowing them to delete list items will help them keep their lists tidy.
 
## Acceptance criteria
- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Notes:
* As noted in [the wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png), activating the “Delete” button should show a confirmation dialog. We want to give users a chance to confirm “destructive” actions.
* It might be tempting to create your own modal dialog component to handle confirmation, but modals are fraught with accessibility concerns, so consider using a simple [JavaScript confirm dialog](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm).

### PR DESCRIPTION
**Pull request by Terieyenike**
## Description

This feature allows users to delete list items to keep their lists clean and tidy.

## Related Issue

Closes #11 

## Acceptance Criteria

- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates
[Screencast from 11-08-2022 06:13:53 PM.webm](https://user-images.githubusercontent.com/68116083/200631451-9fd89a37-83fd-46ee-ae0f-f5b27974e02b.webm)

## Testing Steps / QA Criteria

Please, test the feature's functionality here: https://tcl-49-smart-shopping-list--pr31-tcl-49-delete-items-wgjwbr7t.web.app/

### CODE REVIEW

**Comment by Mama-simba** [comment]
I have checked the functionality on both sides (UI and Firebase) and works perfect. Well done @Terieyenike!

**Comment by thatgirldorian** [comment]
Just tested this and it looks great @Terieyenike ! Nice work! :)

**Review by Mama-simba — COMMENTED** [review]
This happened already on a previous PR. There shouldn't be any edits on husky and the package.json files. I would suggest you to remove the commits where this happened :S

**Comment by Terieyenike** [comment]
> This happened already on a previous PR. There shouldn't be any edits on husky and the package.json files. I would suggest you to remove the commits where this happened :S

Can you help me remove it?

**Comment by Mama-simba** [comment]
Sure! Let's have a call and I will show you how to do it :)

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
-. "$(dirname "$0")/_/husky.sh"
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
 
-npx lint-staged
+npm test
```
Comment (faysvas) [inline .husky/pre-commit]: Why these changed by you or an automated tool?

File: src/components/ListItem.jsx [hunk 2]
```diff
@@ -36,6 +37,18 @@ export function ListItem({ name, items, listToken }) {
...
+		const confirm = window.confirm(`Do you really want to delete ${name}`);
+
+		try {
+			if (confirm) {
+				deleteItem(listToken, id);
+			}
+		} catch (error) {
+			console.log(error);
+		}
+	};
```
Comment (faysvas) [inline src/components/ListItem.jsx]: If you want to catch errors coming from `deleteItem` you should make the `handleDeleteItem` function async and await the result to `deleteItem` like this:
```
	const handleDeleteItem = async () => {
		const confirm = window.confirm(`Do you really want to delete ${name}`);

		try {
			if (confirm) {
				await deleteItem(listToken, id);
			}
		} catch (error) {
			console.log(error);
		}
	};
```

File: src/views/AddItem.jsx [hunk 3]
```diff
@@ -3,30 +3,50 @@ import { Button, FormInput } from '../components';
...
+				addItem(listToken, {
+					itemName,
+					daysUntilNextPurchase,
+				});
+				setMessage(`${itemName} was successfully saved to your shopping list!`);
+			}
 		} catch (error) {
-			setMessage('item not added to db');
+			setMessage('Error: This item not added to the database.');
 		}
```
Comment (faysvas) [inline src/views/AddItem.jsx:46]: These changes don't belong to this issue? What do they do?

File: .husky/pre-commit (hunk 1 above)
Comment (thatgirldorian) [reply .husky/pre-commit]: These were husky changes that could have happened from Git. I've reverted this to what we have in the main branch.

File: src/views/AddItem.jsx (hunk 3 above)
Comment (thatgirldorian) [reply src/views/AddItem.jsx:46]: This is the error message that shows up when a person who doesn't have a token tries to add an item to the list. I just rewrote it to be clearer.

File: src/components/ListItem.jsx (hunk 2 above)
Comment (thatgirldorian) [reply src/components/ListItem.jsx]: Just reworked this, thank you! :)

**Review by faysvas — APPROVED** [review]
Looks good :+1: !

**Comment by pmbanugo** [comment]
@Terieyenike could you pls add the related issue for the PR in the description?

**Comment by Mama-simba** [comment]
@pmbanugo Done!

**Comment by pmbanugo** [comment]
All looks good to me except for the change to package.json, package-lock.json, and a git submodule. They shouldn't be affected by this PR and I don't understand why they're been updated in the recent PRs.

But you can merge if it's not going to cause any problems in your workflow.

**Pull request merged**